test_rclcpp/test/test_executor.cpp

```diff
@@ -28,12 +28,15 @@
 TEST(CLASSNAME(test_executor, RMW_IMPLEMENTATION), recursive_spin_call) {
   rclcpp::executors::SingleThreadedExecutor executor;
   auto node = rclcpp::Node::make_shared("recursive_spin_call");
-  auto timer = node->create_wall_timer(0_s, [&executor]() {
+  auto timer = node->create_wall_timer(0_s,
+      [&executor](rclcpp::timer::TimerBase & timer) {
     ASSERT_THROW(executor.spin_some(), std::runtime_error);
     ASSERT_THROW(executor.spin_once(), std::runtime_error);
     ASSERT_THROW(executor.spin(), std::runtime_error);
     executor.cancel();
-  });
+    timer.cancel();
```

Comment:
Hmm, true, this is redundant.

There is a situation where cancelling the timer without cancelling the executor is useful: when the timer callback is publishing messages for a finite amount of time in-process, the timer can cancel itself after the last publish, wait some amount of time for subscriber to get the message, then cancel the executor...

In this test case, since there is no publish/subscriber, executor.cancel() is enough. Also, it never makes sense to call executor.cancel() before timer.cancel(), as I have done here and in the other test case.